### PR TITLE
first layout of command struct

### PR DESCRIPTION
**Pull request by Knulpinette**
Do you agree with naming ? Structure ? Logic ? I mean, obviously, this is going to change probably as we go overall.

### CODE REVIEW

File: includes/types.h [hunk 1]
```diff
@@ -1,10 +1,38 @@
...
+	bool	delimiter;
+	char	*delim_arg;
+	int		mode; /* OVERWRITE, APPEND */
+	char	**args;
+
+}				t_cmd_table;
+
+typedef	enum	e_redir_mode // for readable instructions when you change the mode according to >>
+{
+	OVERWRITE, // default IS overwrite, right ? (here overwrite = 0)
```
Comment (arieivs) [inline includes/types.h]: I think so yes, that the default is overwrite

File: includes/types.h [hunk 2]
```diff
@@ -17,5 +45,12 @@ typedef enum 	e_error_codes
 	WRONG_ARGC,
 }				t_error;
 
+typedef struct 	s_minishell
+{
+	char		*prompt;
```
Comment (arieivs) [inline includes/types.h]: Why is prompt part of the minishell struct? It's something we decide, so I think we can just have
#define PROMPT '> '

File: srcs/main.c [hunk 3]
```diff
@@ -54,8 +46,8 @@ int	main(int argc, char **argv)
 			return (0);
 		}
 		// validate user input (line)
-		minishell.coconut = line;
-		printf("%s\n", minishell.coconut);
+		minishell.prompt = line;
```
Comment (arieivs) [inline srcs/main.c]: no no no, the line is not going to be the prompt. If you wanna store it, better call it instruction? instruction_line? not sure
The prompt is the "> " argument in readline function above btw

**Review by arieivs — COMMENTED** [review]
Just a small confusion on the prompt, other than that seems great, thank youuuu

File: includes/types.h [hunk 4]
```diff
@@ -1,10 +1,38 @@
 #ifndef TYPES_H
 # define TYPES_H
 
-typedef struct 	s_minishell
+/* PARSING MACROS */
+# define SPACE ' '
+# define PIPE '|'
+# define ENV_VARIABLE '$'
```
Comment (arieivs) [inline includes/types.h]: I forgot before sry, can it be ENV_VAR please please pleaseeeeeeeee

File: includes/types.h (hunk 2 above)
Comment (Knulpinette) [reply includes/types.h]: ok !

File: srcs/main.c (hunk 3 above)
Comment (Knulpinette) [reply srcs/main.c]: Oh, wait I got it. let's put that back to coconut so we know it's something we don't need in the future and should get rid of. I misunderstood something stupid in how you code worked haha.

File: includes/types.h (hunk 4 above)
Comment (Knulpinette) [reply includes/types.h]: haha. ok.

**Pull request merged**